spyder/utils/introspection/manager.py

```diff
@@ -154,7 +183,18 @@ def _handle_timeout(self):
         self.waiting = False
         if self.pending:
             self._finalize(self.pending)
+        elif self.pending_request:
+            info = self.pending_request
+            self.pending_request = None
+            self.send_request(info)
         else:
+            self.timeout_counter += 1
+            if (self.timeout_counter > MAX_TIMEOUTS and
+                    not self.plugins_initialized['jedi']):

```

Comment:
1. I changed it to give the plugins some more time to start instead of restarting then immediately. But maybe something between 5 - 10  should be enough. Should I set `MAX_TIMEOUTS` to 5?
2. Ok :+1: